### ISSUE TITLE
Add MPI testing

### ISSUE THREAD

**Issue by nwu63** [mdolab/pyoptsparse#222]
# Description
We do not have any MPI-related tests for the following:
- [ ] fake COMM object works on single proc
- [ ] optimization in parallel runs fine, with only the root proc running the optimization itself but calling the objective/sensitivity functions on all procs
- [ ] parallel DV/constraint adding works as expected

**Comment by nwu63** [comment]
Superseded by #256
